### PR TITLE
[Standard Transformer] Run metrics server in the same port as GRPC server when running UPI Server

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

Due to limitation in knative (https://github.com/knative/serving/issues/8471) . It's currently not possible to expose prometheus metrics server endpoint in separate port from the grpc port. As such, metrics from standard transformer is not being crapped. 

This PR address that by exposing grpc server and http server (which host prometheus metrics server among other things) in the same port using https://github.com/soheilhy/cmux

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [N.A.] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [N.A.] Updated documentation
- [N.A.] Update Swagger spec if the PR introduce API changes
- [N.A.] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/pkg/transformer/server/grpc/server.go [hunk 1]
```diff
@@ -108,23 +119,44 @@ func (us *UPIServer) Run() {
...
+		us.logger.Info("starting http server")
+		if err := httpServer.Serve(httpLis); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			errCh <- errors.Wrapf(err, "HTTP server failed")
+		}
+	}()
+
+	go func() {
+		us.logger.Info(fmt.Sprintf("serving at port: %s", us.opts.GRPCPort))
+		if err := m.Serve(); err != nil && !errors.Is(err, cmux.ErrListenerClosed) {
+			errCh <- errors.Wrapf(err, "server failed")
```
Comment (tiopramayudi) [inline api/pkg/transformer/server/grpc/server.go:139]: nit: "instrumentation server failed"

Comment (tiopramayudi) [inline api/pkg/transformer/server/grpc/server.go:139]: nit: "cmux failed" ?

File: api/pkg/transformer/server/grpc/server.go [hunk 2]
```diff
@@ -108,23 +119,44 @@ func (us *UPIServer) Run() {
...
 
+	us.logger.Info("shutting down standard transformer")
 	if err := us.conn.Close(); err != nil {
 		us.logger.Error(fmt.Sprintf("failed to close connection %v", err))
 	}
 	us.logger.Info("closed connection to model prediction server")
+
 	grpcServer.GracefulStop()
+	us.logger.Info("stopped grpc server")
+	_ = httpServer.Shutdown(context.Background())
```
Comment (tiopramayudi) [inline api/pkg/transformer/server/grpc/server.go:158]: should we log if shutdown is throwing error?

File: scripts/e2e/deploy-merlin.sh [hunk 3]
```diff
@@ -23,7 +23,6 @@ install_mlp() {
     --set mlp.ingress.enabled=true \
     --set mlp.ingress.class=istio \
     --set mlp.ingress.host=mlp.mlp.${INGRESS_HOST} \
-    --set mlp.ingress.path="/*" \
```
Comment (tiopramayudi) [inline scripts/e2e/deploy-merlin.sh:26]: why do we remove this?

Comment (pradithya) [reply scripts/e2e/deploy-merlin.sh:26]: Because https://github.com/gojek/mlp/pull/67 the path is `PrefixType` and shouldn't contain wildcard

**Review by ariefrahmansyah — APPROVED** [review]

**Pull request merged**